File: DsWorkJava/dswork.core/dswork/core/mybatis/dialect/SQLServer2000Dialect.java
package dswork.core.mybatis.dialect;

/**
 * Dialect for SQLServer2000，查询的sql必须存在ID列，且翻页以ID顺序排序
 * @author skey
 */
public class SQLServer2000Dialect extends Dialect
{
	/**
	 * 是否支持分页，limit和offset
	 * @return boolean
	 */
	public boolean supportsLimitOffset()
	{
		return true;
	}
	
	/**
	 * 将sql变成分页sql
	 * @param sql 原始sql语句
	 * @param offset 跳过行数
	 * @param limit 返回行数
	 * @return String
	 */
	public String getLimitString(String sql, int offset, int limit)
	{
		StringBuilder sb = new StringBuilder(sql.length() + 150);
		sb.append("select top ").append(limit).append(" n_.* from (")
			.append("select top ").append(limit).append(" m_.* from (")
				.append("select top ").append(offset + limit).append(" t_.* from (")
				.append(sql)
				.append(") t_ order by t_.ID")
			.append(") m_ order by m_.ID desc")
		.append(") n_ order by n_.ID");
		return sb.toString();
	}
}

File: DsWorkNet/Dswork.Core/Mybaits/Dialect/SQLServer2000Dialect.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Dswork.Core.Mybaits.Dialect
{
	/// <summary>
	/// Dialect for SQLServer2000，查询的sql必须存在ID列，且翻页以ID顺序排序
	/// </summary>
	public class SQLServer2000Dialect:Dialect
	{
		/// <summary>
		/// 是否支持分页，limit和offset
		/// </summary>
		/// <returns>Boolean</returns>
		public override Boolean SupportsLimitOffset()
		{
			return true;
		}

		/// <summary>
		/// 将sql变成分页sql
		/// </summary>
		/// <param name="sql">原始sql语句</param>
		/// <param name="offset">跳过行数</param>
		/// <param name="limit">返回行数</param>
		/// <returns>String</returns>
		public override String GetLimitString(String sql, int offset, int limit)
		{
			StringBuilder sb = new StringBuilder(sql.Length + 150);
			sb.Append("select top ").Append(limit).Append(" n_.* from (")
				.Append("select top ").Append(limit).Append(" m_.* from (")
					.Append("select top ").Append(offset + limit).Append(" t_.* from (")
					.Append(sql)
					.Append(") t_ order by t_.ID")
				.Append(") m_ order by m_.ID desc")
			.Append(") n_ order by n_.ID");
			return sb.ToString();
		}
	}
}
